pkg/autogen/rule.go

```diff
@@ -321,17 +321,6 @@ func updateGenRuleByte(pbyte []byte, kind string) (obj []byte) {
 	return obj
 }
 
-func updateRestrictedFields(pbyte []byte, kind string) (obj []byte) {
```

Comment:
Why is it removed?

Reply:
The function is not being used anymore. `updateGenRuleByte()` is being used in all cases.

Reply:
It was renamed to `updateGenRuleByte`.